core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -280,9 +285,53 @@ protected ClassLoader createClassLoader(List<File> paths, ClassLoader parent, At
     }
 
     /**
+     * Get the list of all plugins that have ever been detached from Jenkins core.
```

Comment:
I had to read this a few times to work out what you mean by detached plugin.
Because I have context I know what you men - but not sure it is completely obvious to someone new to the code.

that is a detached plugin is one that
a) was an existing plugin that at some time previously bundled with the Jenkins war file
or
b) was previous code in jenkins core that was split to a separate-plugin (but may not have ever been bundled in a jenkins war file - i.e. it gets split after this 2.0 update)